operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -0,0 +1,284 @@
+    return hasPermissionForResource(
+        decisionId, AuthorizationResourceType.DECISION_DEFINITION, identityPermission);
+  }
+
+  /**
+   * hasPermissionForResource
+   *
+   * @return true if the user has the given permission for the resource
+   */
+  public boolean hasPermissionForResource(
+      final String resourceId,
+      final AuthorizationResourceType resourceType,
+      final IdentityPermission identityPermission) {
+    if (!permissionsEnabled()) {
+      return true;
+    }
+    if (!isAuthorized()) {
+      return false;
+    }
+    if (identityPermission == null) {
+      throw new IllegalStateException("Identity permission cannot be null");
+    }
+    final PermissionType permissionType = getPermission(identityPermission);
+
+    return isAuthorizedFor(resourceId, resourceType, permissionType);
+  }
+
+  /**
+   * getProcessesWithPermission
+   *
+   * @return processes for which the user has the given permission; the result matches either all
+   *     processes, or a list of bpmnProcessId
+   */
+  public ResourcesAllowed getProcessesWithPermission(final IdentityPermission identityPermission) {
+    return getResourcesWithPermission(
+        AuthorizationResourceType.PROCESS_DEFINITION, identityPermission);
+  }
+
+  /**
+   * getDecisionsWithPermission
+   *
+   * @return decisions for which the user has the given permission; the result matches either all
+   *     decisions, or a list of decisionId
+   */
+  public ResourcesAllowed getDecisionsWithPermission(final IdentityPermission identityPermission) {
+    return getResourcesWithPermission(
+        AuthorizationResourceType.DECISION_DEFINITION, identityPermission);
+  }
+
+  /**
+   * getResourcesWithPermission
+   *
+   * @return resources for which the user has the given permission; the result matches either all
+   *     resources, or a list of resourceIds
+   */
+  public ResourcesAllowed getResourcesWithPermission(
+      final AuthorizationResourceType resourceType, final IdentityPermission identityPermission) {
+    if (!permissionsEnabled()) {
+      return ResourcesAllowed.all();
+    }
+    if (!isAuthorized()) {
+      return ResourcesAllowed.withIds(Set.of());
+    }
+    if (identityPermission == null) {
+      throw new IllegalStateException("Identity permission cannot be null");
+    }
+    final PermissionType permissionType = getPermission(identityPermission);
+    final Authorization authorization = new Authorization(resourceType, permissionType);
+    final SecurityContext securityContext = getSecurityContext(authorization);
+    final List<String> ids = authorizationChecker.retrieveAuthorizedResourceKeys(securityContext);
+
+    return ResourcesAllowed.withIds(new LinkedHashSet<>(ids));
+  }
+
+  /**
+   * @return true if permissions checks are enabled
+   */
+  public boolean permissionsEnabled() {
+    return securityConfiguration.getAuthorizations().isEnabled();
+  }
+
+  private boolean isAuthorized() {
```

Comment:
`isAuthenticated()` ?
I expect that the call is not authenticated, is detected beforehand. But maybe we miss the OIDC cases handing, isn't?

Reply:
This method is actually subject to change, after we have the implementation from Identity. I am not sure about how OIDC is supposed to work though.